share/man/man9/cdefs.9

```diff
@@ -0,0 +1,297 @@
+things about the build environment.
+.Bl -column ".Sy Macro" ".Sy Description"
+.It Sy Macro Ta Sy Description
+.It Sy __LONG_LONG_SUPPORTED Ta Variables may be declared
+.Dq long long .
+This is defined for C99 or newer and C++ environments.
+.It Sy __STDC_LIMIT_MACROS Ta
+.It Sy __STDC_CONSTANT_MACROS Ta
+.El
+.Ss Nice to have macros
+.Bl -column ".Sy Macro" ".Sy Description"
+.It Sy Macro Ta Sy Description
+.It Sy __offsetof(type,field) Ta
+.It Sy __rangeof(type,start,end) Ta
+.It Sy __containerof(x,s,m) Ta
+.El
+.Ss ID Strings
+This section is deprecated, but is kept around because too much contrib software
+still uses these.
+.Bl -column ".Sy Macro" ".Sy Description"
+.It Sy Macro Ta Sy Description
+.It Sy __IDSTRING(name,string) Ta
+.It Sy __FBSDID(s) Ta
+.It Sy __RCSID(s) Ta
+.It Sy __RCSID_SOURCE(s) Ta
+.It Sy __SCCSID(s) Ta
+.It Sy __COPYRIGHT(s) Ta
+.El
+.Sh Supported C Environments
+FreeBSD supports a number C standard environments.
+Selection of the language dialect is a compiler dependent command line option,
+though it is usually
+.Fl cstd=XX
+where XX is the standard you desire, like c89 or c23.
+FreeBSD provides a number of selection macros to control visibility of symbols.
+Please see the section on Selection Macros for the specifics.
+.Bl -tag -compact
+.It K \*(Am R
+Pre-ansi Kernighan and Ritchie C.
+Sometimes called
+.Dq knr
+or
+.Dq C78
+to distinguish it from newer standards.
+Support for this compilation environment is dependent on compilers supporting
+this configuration.
+Most of the old forms of C have been deprecated or removed in
+ISO/IEC 9899:2024 (“ISO C23”) and newer compilers may require many warnings to
+be suppressed to active this mode.
+.It St -ansiC
+.Dv __STDC__
+is defied, however
+.Dv __STDC_VERSION__
+is not.
+.Pp
+Strict environment selected with
+.Dv _ANSI_SOURCE .
+.It St -isoC-99
+.Dv __STDC_VERSION__ = 199901L
+.Pp
+Strict environment selected with
+.Dv _C99_SOURCE .
+.It St -isoC-2011
+.Dv __STDC_VERSION__ = 201112L
+.Pp
+Strict environment selected with
+.Dv _C11_SOURCE .
+.It ISO/IEC 9899:2018 (“ISO C17”)
+.Dv __STDC_VERSION__ = 201710L
+.Pp
+Strict environment selected with
+.Dv _C11_SOURCE
+since there are no new C17 only symbols or macros.
+.Pp
+This version of the standard did not introduce any new features, only made
+minor, technical corrections.
+.It ISO/IEC 9899:2024 (“ISO C23”)
+.Dv __STDC_VERSION__ = 202311L
+Strict environment selected with
+.Dv _C23_SOURCE
+though this is not yet implemented.
+.El
+.Pp
+For more information on C standards, see
+.Xr c 7 .
+.Ss Programming Environment Selection Macros
+By defining the macros as outlined below, the compilation environment requests
+that only symbols and macros visible to the described environment be defined by
+the standard system header file inclusion.
+Including non-standard header files may result in macros and symbols not
+described in the environment being defined.
```

Comment:
```suggestion
Including non-standard header files may result in definition of macros and
symbols that are not described in the environment.
```
Did I understand this correctly?

Reply:
No. Normally when you define the selection macro, only macros, functions, structures and variables defined by what's selected will be defined (except things in the implementation space). But this guarantee is only true if you include files defined by that standard. If you include other system header files, all bets are off. So it's better to say these definitions will be in the compilation environment, despite not being defined by that requested standard.

Reply:
I'll work on a better wording though.